### ISSUE TITLE
don't re-run parser for each project switch (tabbing)

### ISSUE THREAD

**Issue by falafeljan** [ununu-p2p/ink-desktop#61]
switching between project tabs now re-runs the parser for each project. maybe use react fiber/suspense and/or cache the component state inbetween changes?

**Comment by shresthagrawal** [comment]
@falafeljan if I understood your problem statement correctly, we already solve it with the parser manager! we only parse the file once and store it in the map.

**Comment by falafeljan** [comment]
that's correct, but the parser is still re-run on each tab switch. maybe it's not actually re-run, but nevertheless each project receives a full re-render on each tab switch. we could solve this probably by storing state externally, with something like redux (https://redux.js.org/) or mobx (https://mobx.js.org/).

**Comment by shresthagrawal** [comment]
@falafeljan I think the tabs are quite smooth in my system even if it is re rendered.  Do you feel the lag in your system? if not much may be can keep this for later sprints?
